plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -226,6 +254,36 @@ func CreateSampleID(tags map[string]string) SampleID {
 	return SampleID(strings.Join(pairs, ","))
 }
 
+func (p *PrometheusClient) createFamily(pvt prometheus.ValueType, point telegraf.Metric, sample *Sample, mname string, sampleID SampleID) {
```

Comment:
This function doesn't always do what it says it does "createFamily", and its does a lot more.  I think it probably needs to be something like `func addMetricFamily(name string) MetricFamily` and only do the first half and then another function to `addSample(`.